pkg/apis/v1/ec2nodeclass_validation_cel_test.go

```diff
@@ -604,6 +604,15 @@ var _ = Describe("CEL/Validation", func() {
 			nc.Spec.AMISelectorTerms = []v1.AMISelectorTerm{{Alias: "ubuntu@latest"}}
 			Expect(env.Client.Create(ctx, nc)).ToNot(Succeed())
 		})
+		DescribeTable(
```

Comment:
Can we add a test that would have caught the regex failure that we caught in this PR?

Reply:
It wasn't really a regex failure, calling it a mistake was a mistake. The mistake was really in this PR, assuming version already had a non-zero length. For non-windows AMIs, version validation is really entirely done at runtime.

Reply:
Still seems like we should add a test, no?

Reply:
Sure, it makes sense to add a test to ensure this validation invariant holds. I thought you meant there was some other existing failure we should add a test for. I've added it.